src/Psalm/Internal/Analyzer/Statements/Expression/Assignment/ArrayAssignmentAnalyzer.php

```diff
@@ -706,7 +706,6 @@ private static function updateTypeWithKeyValues(
                     }
                 } elseif ($type instanceof TNonEmptyList
                     && $key_value instanceof Type\Atomic\TLiteralInt
-                    && ($key_value->value === $key_value->value || $key_value->value < $type->count)
```

Comment:
while the lhs of that or looks really redundant, the rhs does not immediately strike as useless

Reply:
yep, the first commit deleted only the first part, but this is a "or". So `true || A` is true. Deleting only the first part changed the behaviour, and indeed, CI failed as a result.